source/material_model/rheology/drucker_prager.cc

```diff
@@ -57,13 +58,33 @@ namespace aspect
                                              const double angle_internal_friction,
                                              const double pressure,
                                              const double effective_strain_rate,
-                                             const double max_yield_stress) const
+                                             const double max_yield_stress,
+                                             const bool use_plastic_damper,
+                                             const double damper_viscosity,
+                                             const double pre_yield_viscosity) const
       {
         const double yield_stress = compute_yield_stress(cohesion, angle_internal_friction, pressure, max_yield_stress);
 
         const double strain_rate_effective_inv = 1./(2.*effective_strain_rate);
 
-        return yield_stress * strain_rate_effective_inv;
+        double plastic_viscosity = yield_stress * strain_rate_effective_inv;
+
+        if (use_plastic_damper == true)
+          {
+            const double total_stress = ( yield_stress + ( 2. * damper_viscosity * effective_strain_rate ) ) /
+                                        ( 1 + ( damper_viscosity / pre_yield_viscosity ) );
+
+            const double pre_yield_strain_rate = total_stress / ( 2. * pre_yield_viscosity);
+
+            const double plastic_strain_rate = effective_strain_rate - pre_yield_strain_rate;
+
+            plastic_viscosity = yield_stress / (2.*plastic_strain_rate) + damper_viscosity;
+
+            // Effective viscosity
+            plastic_viscosity = 1. / (1./plastic_viscosity + 1./pre_yield_viscosity);
```

Comment:
Thinking some more on our conversation, I think pre_yield_viscosity should have some effect whether or not the plastic damper is used. In other words, this function should produce identical output for the inputs:
`use_plastic_damper = false`
`damper_viscosity = 1e8`
`pre_yield_viscosity = 1e10` (for example)
or
`use_plastic_damper = true`
`damper_viscosity = 0.`
`pre_yield_viscosity = 1e10`

Reply:
@bobmyhill - that was the original intent, but unfortunately I don't see a way for this to work without breaking backwards compatibility. 

When `use_plastic_damper = true`, the plastic viscosity uses the plastic strain rate (`plastic_strain_rate`) rather than the total strain rate (standard method) to calculate the plastic viscosity.

Long-term, I would be in favor of only using the damper method where a unique plastic strain is calculated, but that will be a major switch unless there is someway around it I'm not seeing.

Reply:
I'm not sure I understand. You've added new variables to the function, so I think backwards compatibility is already broken.

My suggestion is just to move 
`plastic_viscosity = 1. / (1./plastic_viscosity + 1./pre_yield_viscosity);` outside of the if clause, and have pre_yield_viscosity = std::numeric_limits<double>::infinity() as a default value. Is there some reason this wouldn't work?

Reply:
Oh, I think I understand now. Is it true that pre_yield_viscosity is only useful when the viscous damper is active? If that's true, then the current code design makes sense.

Reply:
@bobmyhill - sorry for the confusion through all that! Yes, ` pre_yield_viscosity` is only used when the damper is active. They are not intrinsically related to other, but the averaging with         `pre_yield_viscosity` is used by Duretz et al. We may want to rethink about the the default assumptions later on and break backwards compatibility in multiple places in one fell swoop.